src/main/webapp/script.js

```diff
@@ -64,6 +64,52 @@ function createPostElement(post) {
     return postElement;
 }
 
+function validateLocation() {
```

Comment:
that's actually a great point! I was wading through a bunch of resources, so instead of working on google cloud shell I actually ended up working on vscode (just so I could check my html faster without having to deploy the app). Then I pretty much copied and pasted from my own vscode into the cloud shell, but I think you make a great point that the two js files can actually be merged. Working on that now!

Reply:
actually please disregard my previous comment. What you said is definitely a great idea, but the main reason why the JS was split into two files was because of the onAuthStateChanged function. 

Originally I meant to just have one onAuthStateChanged function. But when I tried "if (user){redirect main.html} else{redirect to log in}" format (which was in fact the recommended way in the documentation), while the user's not logged in (which is the default state), it would just always go to the else case and keep loading the index.html without giving users a chnace to log in. I tried to find ways to fix this, but when I couldn't, I just resorted to having two files of JS since the two different cases/functions both require onAuthStateChanged event. 

Do you happen to know how I could fix the code so I can just deal with both login/ logout scenarios with just one onAuthStateChanged function?